#68 - Issue 26 improvement in recycler view performance.

Ivan2001otp opened the pull request:
<!-- If your PR fixes an open issue, use `Closes #999` to link your PR with the issue. #999 stands for the issue number you are fixing -->

## Fixes Issue
Fixed the issue , by implementing listAdapter and DiffUtil.ItemCallback class.




## Note to reviewers
Some changes have been observed in gradle file and other file stuff, this happened as I pulled the changes from the master branch of project's repository. 

<!-- Add notes to reviewers if applicable -->

Ivan2001otp commented:
Please let me know,If there is any mistake.Thank you.

Vaibhav2002 commented on app/src/main/java/com/itssuryansh/taaveez/itemAdapter.kt:
```diff
@@ -55,7 +59,22 @@ class itemAdapter(private val items: ArrayList<NotesEntity>,
+       }
+
+       override fun areContentsTheSame(oldItem: NotesEntity, newItem: NotesEntity): Boolean {
+            return when{
```
return oldItem == newItem

Vaibhav2002 commented on app/src/main/java/com/itssuryansh/taaveez/itemAdapter.kt:
```diff
@@ -2,18 +2,20 @@ package com.itssuryansh.taaveez
 
 
-class itemAdapter(private val items: ArrayList<NotesEntity>,
+class itemAdapter(
```
Use PascalCase for class names

Vaibhav2002 replied:
@Ivan2001otp

Vaibhav2002 commented on app/src/main/java/com/itssuryansh/taaveez/itemAdapter.kt:
```diff
@@ -31,9 +33,11 @@ class itemAdapter(private val items: ArrayList<NotesEntity>,
     override fun onBindViewHolder(holder: ViewHolder, position: Int)  {
         val context = holder.itemView.context
-        val item = items[position]
+        val item = getItem(position)
```
use `currentList[position]`

Vaibhav2002 replied:
@Ivan2001otp make the requested changes bro

Vaibhav2002 reviewed (changes requested):
Please make the necessary changes

Vaibhav2002 commented:
@Ivan2001otp you could have added more commits to the previous PR, would have been easier to track the requested changes

Nice work by the way.
Hope you have learned something new.

Ivan2001otp commented:
Thank you sir.

On Sat, Jun 10, 2023 at 6:36 PM Vaibhav Jaiswal ***@***.***>
wrote:

> @Ivan2001otp <https://github.com/Ivan2001otp> you could have added more
> commits to the previous PR, would have been easier to track the requested
> changes
>
> Nice work by the way.
> Hope you have learned something new.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Taaveez/Taaveez-android/pull/70#issuecomment-1585656620>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWYXRHTRXDJHBZ7JPFOMNZ3XKRWOBANCNFSM6AAAAAAZBUB7RQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Vaibhav2002 commented:
> Thank you sir.
> […](#)
> On Sat, Jun 10, 2023 at 6:36 PM Vaibhav Jaiswal ***@***.***> wrote: @Ivan2001otp <https://github.com/Ivan2001otp> you could have added more commits to the previous PR, would have been easier to track the requested changes Nice work by the way. Hope you have learned something new. — Reply to this email directly, view it on GitHub <[#70 (comment)](https://github.com/Taaveez/Taaveez-android/pull/70#issuecomment-1585656620)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AWYXRHTRXDJHBZ7JPFOMNZ3XKRWOBANCNFSM6AAAAAAZBUB7RQ> . You are receiving this because you were mentioned.Message ID: ***@***.***>

Are the changes done?

Ivan2001otp commented:
I have updated it,Already. I

On Fri, Jun 16, 2023, 2:40 PM Vaibhav Jaiswal ***@***.***>
wrote:

> @Ivan2001otp <https://github.com/Ivan2001otp> please make the changes
> requested, so we can merge this PR
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Taaveez/Taaveez-android/pull/70#issuecomment-1594370513>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWYXRHTTD4ZKNCVEEMQEFX3XLQPGZANCNFSM6AAAAAAZBUB7RQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Ivan2001otp commented:
I will do the changes in the pr.

On Fri, Jun 16, 2023, 2:52 PM Immanuel Dsouza ***@***.***> wrote:

> I have updated it,Already. I
>
> On Fri, Jun 16, 2023, 2:40 PM Vaibhav Jaiswal ***@***.***>
> wrote:
>
>> @Ivan2001otp <https://github.com/Ivan2001otp> please make the changes
>> requested, so we can merge this PR
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Taaveez/Taaveez-android/pull/70#issuecomment-1594370513>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AWYXRHTTD4ZKNCVEEMQEFX3XLQPGZANCNFSM6AAAAAAZBUB7RQ>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>

Ivan2001otp commented on app/src/main/java/com/itssuryansh/taaveez/itemAdapter.kt:
```diff
@@ -2,18 +2,20 @@ package com.itssuryansh.taaveez
 
-class itemAdapter(private val items: ArrayList<NotesEntity>,
+class itemAdapter(
+    /*private var oldItems: ArrayList<NotesEntity>,*/
```
Yeah i will .. Right now I will fix it.

Vaibhav2002 replied:
Mark this as resolved

Ivan2001otp commented:
Sir, now you can check,i have removed the commented line you mentioned.

On Fri, Jun 16, 2023, 2:54 PM Immanuel Dsouza ***@***.***> wrote:

> I will do the changes in the pr.
>
> On Fri, Jun 16, 2023, 2:52 PM Immanuel Dsouza ***@***.***>
> wrote:
>
>> I have updated it,Already. I
>>
>> On Fri, Jun 16, 2023, 2:40 PM Vaibhav Jaiswal ***@***.***>
>> wrote:
>>
>>> @Ivan2001otp <https://github.com/Ivan2001otp> please make the changes
>>> requested, so we can merge this PR
>>>
>>> —
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/Taaveez/Taaveez-android/pull/70#issuecomment-1594370513>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/AWYXRHTTD4ZKNCVEEMQEFX3XLQPGZANCNFSM6AAAAAAZBUB7RQ>
>>> .
>>> You are receiving this because you were mentioned.Message ID:
>>> ***@***.***>
>>>
>>

Vaibhav2002 commented:
@Ivan2001otp what about the remaining changes. I don't see them updated, please check the requested changes above

Vaibhav2002 commented:
> I have updated it,Already. I
> […](#)
> On Fri, Jun 16, 2023, 2:40 PM Vaibhav Jaiswal ***@***.***> wrote: @Ivan2001otp <https://github.com/Ivan2001otp> please make the changes requested, so we can merge this PR — Reply to this email directly, view it on GitHub <[#70 (comment)](https://github.com/Taaveez/Taaveez-android/pull/70#issuecomment-1594370513)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AWYXRHTTD4ZKNCVEEMQEFX3XLQPGZANCNFSM6AAAAAAZBUB7RQ> . You are receiving this because you were mentioned.Message ID: ***@***.***>

I think you might have forgot to push the changes, i cannot see the commit or the changes

Vaibhav2002 commented on app/src/main/java/com/itssuryansh/taaveez/Notes.kt:
```diff
@@ -69,17 +85,54 @@ class Notes : AppCompatActivity() {
+                    Log.e("ray", "onCreate submit list: $it")
+
+                    Log.e("ray", "onCreate size status: ${it.size}")
+                    Log.e("ray", "onCreate list-empty status: ${it.isEmpty()}")*/
```
remove these comments

Vaibhav2002 replied:
mark this as resolved

Ivan2001otp commented:
Fixed. Removed unnecessary comments .

On Fri, Jun 16, 2023 at 3:54 PM Vaibhav Jaiswal ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In app/src/main/java/com/itssuryansh/taaveez/Notes.kt
> <https://github.com/Taaveez/Taaveez-android/pull/70#discussion_r1232069482>
> :
>
> > @@ -69,17 +85,54 @@ class Notes : AppCompatActivity() {
>          }
>
>          lifecycleScope.launch {
> -            NotesDao.fetchAllNotes().collect {
> -                val list = ArrayList(it)
> -                setupListOfDateINtoRecycleVIew(list, NotesDao)
> +                NotesDao.fetchAllNotes().collect {
> +                    /*
> +                    Log.e("ray", "onCreate submit list: $it")
> +
> +                    Log.e("ray", "onCreate size status: ${it.size}")
> +                    Log.e("ray", "onCreate list-empty status: ${it.isEmpty()}")*/
>
> remove these comments
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Taaveez/Taaveez-android/pull/70#pullrequestreview-1483172929>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWYXRHUVYSP6K6NXBHID7BDXLQX4BANCNFSM6AAAAAAZBUB7RQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Ivan2001otp commented:
Please specify what exact changes I have to make in itemAdapter.kt.

On Tue, Jun 20, 2023 at 12:20 PM Vaibhav Jaiswal ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In app/src/main/java/com/itssuryansh/taaveez/itemAdapter.kt
> <https://github.com/Taaveez/Taaveez-android/pull/70#discussion_r1234813315>
> :
>
> >      override fun onBindViewHolder(holder: ViewHolder, position: Int)  {
>          val context = holder.itemView.context
> -        val item = items[position]
> +        val item = getItem(position)
>
> @Ivan2001otp <https://github.com/Ivan2001otp> make the requested changes
> bro
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Taaveez/Taaveez-android/pull/70#discussion_r1234813315>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWYXRHTS7QNFNF65KV22YBDXMFB4LANCNFSM6AAAAAAZBUB7RQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Vaibhav2002 commented:
> Please specify what exact changes I have to make in itemAdapter.kt.
> […](#)
> On Tue, Jun 20, 2023 at 12:20 PM Vaibhav Jaiswal ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In app/src/main/java/com/itssuryansh/taaveez/itemAdapter.kt <[#70 (comment)](https://github.com/Taaveez/Taaveez-android/pull/70#discussion_r1234813315)> : > override fun onBindViewHolder(holder: ViewHolder, position: Int) { val context = holder.itemView.context - val item = items[position] + val item = getItem(position) @Ivan2001otp <https://github.com/Ivan2001otp> make the requested changes bro — Reply to this email directly, view it on GitHub <[#70 (comment)](https://github.com/Taaveez/Taaveez-android/pull/70#discussion_r1234813315)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AWYXRHTS7QNFNF65KV22YBDXMFB4LANCNFSM6AAAAAAZBUB7RQ> . You are receiving this because you were mentioned.Message ID: ***@***.***>

Bro please open https://github.com/Taaveez/Taaveez-android/pull/70 and see all the changes requested
